### ISSUE TITLE
Different results in train/validation and predicition

### ISSUE THREAD

**Issue by cisi17** [BVLC/caffe#5764]
Hello,

I have implemented my own neural network with my own dataset.

Firstly. I train and test my network using transfer learning. I have achieved 95% train/test accuaracy:

      /usr/local/micaffe1/bin/caffe train --solver=/home/ubuntu/example/modelcsl/solver_csl.prototxt --weights /home/ubuntu/example/bvlc_reference_caffenet.caffemodel 2>&1 | tee /ubuntu/example/modelcsl/resultados_TL.log

But when I am going to do a prediction, the precision is over 70%:

      net = caffe.Net('/home/marcelo/ejemplosCaffe1/legImgLaser/modelcsl/caffenet_deploy_csl.prototxt',
                '/home/marcelo/ejemplosCaffe1/legImgLaser/modelcsl_iter_500.caffemodel',
                caffe.TEST)
......

       net.blobs['data'].data[...] = transformer.preprocess('data', img)
       out = net.forward()
       pred_probas = out['prob']
.......

I think that the deploy.prototxt and the train_test.prototxt have got the same net. And the preprocess is the same.

Could you help me?

Thank you.

**Comment by Noiredd** [comment]
For one thing, you shouldn't be directly comparing precision to accuracy, as they are different measures. Another is that a discrepancy between train/val and test results is common. Perhaps your training dataset contained very "easy" examples and the test set had all the "hard" ones? You should cross-validate your model before drawing any conclusions.

Please post usage, installation, or modeling questions, or other requests for help to the [caffe-users list][1] instead of Issues. This helps developers maintain a clear, uncluttered, and efficient view of the state of Caffe. Please read the [guidelines for contributing][2] before submitting an issue or a pull request.

 [1]: https://groups.google.com/forum/#!forum/caffe-users
 [2]: https://github.com/BVLC/caffe/blob/master/CONTRIBUTING.md
